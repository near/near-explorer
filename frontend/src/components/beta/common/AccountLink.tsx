import * as React from "react";
import Link from "../../utils/Link";
import { shortenString } from "../../../libraries/formatting";
import { styled } from "../../../libraries/styles";

const AccountLinkWrapper = styled("a", {
  whiteSpace: "nowrap",
});

export interface Props {
  accountId: string;
}

const AccountLink: React.FC<Props> = React.memo(({ accountId }) => {
  return (
    <Link href={`/beta/accounts/${accountId}`} passHref>
      <AccountLinkWrapper>{shortenString(accountId)}</AccountLinkWrapper>
    </Link>
  );
});

export default AccountLink;
