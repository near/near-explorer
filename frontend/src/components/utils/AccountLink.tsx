import Link from "next/link";

export interface Props {
  accountId: string;
  children?: React.ReactNode;
}

export default ({ accountId, children }: Props) => (
  <Link href="/accounts/[id]" as={`/accounts/${accountId}`}>
    <a className="account-link">{children || `@${accountId}`}</a>
  </Link>
);
