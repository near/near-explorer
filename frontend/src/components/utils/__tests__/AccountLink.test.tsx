import renderer from "react-test-renderer";

import AccountLink from "../AccountLink";

describe("<AccountLink />", () => {
  it("renders", () => {
    expect(renderer.create(<AccountLink accountId="test" />)).toMatchSnapshot();
  });
});
