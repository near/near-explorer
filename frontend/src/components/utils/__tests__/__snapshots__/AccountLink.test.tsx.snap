// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountLink /> renders 1`] = `
<a
  className="account-link"
  href="/accounts/test"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  @test
</a>
`;
